Documentation of epsilon parameter (#330)
scikit-learn-contrib/hdbscan: docs/how_to_use_epsilon.rst

Combining HDBSCAN\* with DBSCAN 
=============================

While DBSCAN needs a minimum cluster size *and* a distance threshold epsilon as user-defined input parameters, 
HDBSCAN\* is basically a DBSCAN implementation for varying epsilon values and therefore only needs the minimum cluster size as single input parameter.
The ``'eom'`` (Excess of Mass) cluster selection method then returns clusters with the best stability over epsilon.

> Unlike DBSCAN, this allows to find clusters of variable densities without having to choose a suitable distance treshold first.

Reviewer: ```suggestion
Unlike DBSCAN, this allows to it find clusters of variable densities without having to choose a suitable distance threshold first.
```

However, there are cases where we could still benefit from the use of an epsilon threshold.

For illustration, see this map with GPS locations, representing recorded pick-up and drop-off locations for customers of a ride pooling provider.
The largest (visual) data cluster can be found around the train station. Smaller clusters are placed along the streets, depending on the requested location

> in form of a postal address or point of interest. Since we are considering a door-to-door system where customers are not bound to collective pick-up or

Reviewer: ```suggestion
in the form of a postal address or point of interest. Since we are considering a door-to-door system where customers are not bound to collective pick-up or
```

but neglects clusters with points that are more than 5 meters apart from each other. 
However, increasing epsilon would result in cluster chains along the streets, especially when working with a larger data set. 

.. image:: images/epsilon_parameter_dbscan.png
	:align: center

Unfortunately, HDBSCAN\* does not produce any better results in this case: while it discovers the clusters that DBSCAN missed, it also returns a very high number of micro-clusters around the train station, 
even though we would prefer one or only few clusters representing this location. We could achieve this by increasing ``min_cluster_size`` or 
the smoothing parameter ``min_samples``, but with the trade-off of losing small clusters in less dense areas or merging them into other clusters 
separated by a relatively large distance.

.. image:: images/epsilon_parameter_HDBSCAN_eom.png
	:align: center
	
This is where the parameter ``cluster_selection_epsilon`` comes into play. The cluster extraction method using this parameter, as described in detail
by `Malzer and Baum <https://arxiv.org/abs/1911.02282>`__, acts like a hybrid between DBSCAN 

> (or, to be precise, DBSCAN*, i.e. DBSCAN without the border points) by extracting DBSCAN results for data partitions

Reviewer: ```suggestion
(or, to be precise, DBSCAN\*, i.e. DBSCAN without the border points) by extracting DBSCAN results for data partitions
```
